zhandao/zero-rails
# Refactor test doc

Commit 83280f4, 36d4d92, 026d495 and 9ae468b:
1. Database has been switch to `PostgreSQL`
2. Upgrade ruby to 2.5.0 and rails to 5.1.5
3. CI fixed
4. 分离出 builder_support，修复测试，更新 swagger-ui

Commit d1f538d:
1. ToTree
2. Test for monkey_patches/array
3. Rename dssl funcs for able to run rspec

Commit 254e9c6:
1. 从 view 中抽离出 render_error helper method

Commit d51b028..068902c:
1. Perfect the tests.

Commit 674fa53:
1. Upgrade `activeadmin` to 2.0
2. Use `ikkiuchi/generators`
3. MDocs and Tests
4. Changed (redesign) lots of fields.
5. BuilderSupport: (a) refacor add_when; (b) support when proc
6. `secure_storage` to `has_secure`
7. Remove AOP check permissions

Commit 1ef6829:
1. Rewrote some models
2. monkey_patches require
3. Fix DocConventer.new multiple times

Commit 9394a0d:
1. Use `rubycritic`
2. Fix output status by `unset_http`
3. Upgrade `ZRO` and `ZPP`
4. Fix OutPut

Commit afcba17 and 2d0d67e:
1. Use `error_cls` to refactor Rescuer
2. Perfect `SecureStorage`
3. Unuse `over-commit`
